include/oneapi/dpl/version

```diff
@@ -0,0 +1,42 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_VERSION
+#define _ONEDPL_VERSION
+
+#include "oneapi/dpl/internal/common_config.h"
+
+#if _ONEDPL___cplusplus >= 202002L && __has_include(<version>)
+#    include <version> // The standard C++20 header
+#    define _ONEDPL_STD_FEATURE_MACROS_PRESENT 1
+#    define _ONEDPL_CPP20_RANGES_PRESENT ((__cpp_lib_ranges >= 201911L) && !(__clang__ && __clang_major__ < 16))
+#else
+#    define _ONEDPL_STD_FEATURE_MACROS_PRESENT 0
+#    define _ONEDPL_CPP20_RANGES_PRESENT 0
+#endif
+
+// The library version
+#define ONEDPL_VERSION_MAJOR 2022
+#define ONEDPL_VERSION_MINOR 7
+#define ONEDPL_VERSION_PATCH 0
+
+#ifndef _PSTL_VERSION
+#    define _PSTL_VERSION 14000
+#    define _PSTL_VERSION_MAJOR (_PSTL_VERSION / 1000)
+#    define _PSTL_VERSION_MINOR ((_PSTL_VERSION % 1000) / 10)
+#    define _PSTL_VERSION_PATCH (_PSTL_VERSION % 10)
+#endif
+
+// Feature macros
+#define ONEDPL_HAS_RANDOM_NUMBERS         202409L
```

Comment:
I think I would preferer to change the name schema. To me `HAS` implies `bool`, while we also provide the version of a feature. So, I would name them like
- `ONEDPL_RANDOM_NUMBER_GENERATION`
- `ONEDPL_RANGE_ALGORITHMS`

I can see with some might prefer to use `_HAS_` in the name. It makes a perfect sense for the
`#if ONEDPL_HAS_RANGE_ALGORITHMS` expression.

But for `#if ONEDPL_HAS_RANGE_ALGORITHMS == 202409L` it doesn't look good in my opinion.

Reply:
This is a good point.  I agree with the name change away from `_HAS_`.

Reply:
We have discussed that last year and the schema was agreed back then. It's not ideal but we come to that as a satisfactory compromise, and I am not going to repeat that discussion. My personal preference, for example, was to only use capitals for ONEDPL and the rest to be in lower case, but that was not supported, and even though I still do not like it, it will be used.

A particular reason for `_HAS_` is that it's good for all the feature macros to have a common prefix of suffix, to ease searching those if needed (including by users), and this one is concise and meaningful.

Reply:
I can agree on the extra `_VER` suffix if that helps: `#if ONEDPL_HAS_RANGE_ALGORITHMS_VER == 202409L`. Sounds weirder to me but fine.

Reply:
I see the point about having some shared prefix which isn't merely `ONEDPL_`, and it is hard to think of a better "small" prefix than `_HAS_`... 

Could `_VER_` or `_VERS_` be an alternative small prefix, without a postfix?
I don't have a strong preference between HAS and VER, but my preference would be against a postfix and prefix together just to keep names of reasonable length.

Reply:
`_FEAT_`? :)

Reply:
After this discussion, my preference is probably this order:
`ONEDPL_VER_XXX`
`ONEDPL_FEAT_XXX`
`ONEDPL_HAS_XXX`
`ONEDPL_XXX`
`ONEDPL_HAS_XXX_VER`

But, really any work for me.